core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/ColumnFormatRelation.scala

```diff
@@ -169,7 +169,8 @@ class BaseColumnFormatRelation(
   override def cachedBatchAggregate(batch: CachedBatch): Unit = {
     // if number of rows are greater than columnBatchSize then store
     // otherwise store locally
-    if (batch.numRows >= Constant.COLUMN_MIN_BATCH_SIZE) {
+    if (batch.numRows >= Constant.COLUMN_MIN_BATCH_SIZE ||
+        java.lang.Boolean.getBoolean("forceFlush")) {
```

Comment:
Yes. Our scala tests uses a batch size of 3. if we use this flag it will always insert in cached batch table. Later we can validate number of rows which landed in column table.